### PR TITLE
Implement #loadProgram ByteArray

### PR DESCRIPTION
**Pull request by anvacaru**
Part of : https://github.com/runtimeverification/firefly/issues/1150

### CODE REVIEW

File: evm.md [hunk 1]
```diff
@@ -1461,15 +1466,14 @@ For each `CALL*` operation, we make a corresponding call to `#call` and a state-
          </k>
 
     rule <k> #mkCreate ACCTFROM ACCTTO VALUE INITCODE
-          => #initVM ~> #execute
+          => #loadProgram INITCODE
+          ~> #initVM ~> #execute
```
Comment (ehildenb) [inline evm.md:1470]:
```suggestion
          => #loadProgram INITCODE ~> #initVM ~> #execute
```

File: evm.md [hunk 2]
```diff
@@ -1292,7 +1292,8 @@ The various `CALL*` (and other inter-contract control flow) operations will be d
          </k>
 
     rule <k> #mkCall ACCTFROM ACCTTO ACCTCODE BYTES APPVALUE ARGS STATIC:Bool
-          => #initVM ~> #precompiled?(ACCTCODE, SCHED) ~> #execute
+          => #loadProgram BYTES
+          ~> #initVM ~> #precompiled?(ACCTCODE, SCHED) ~> #execute
```
Comment (ehildenb) [inline evm.md:1296]:
```suggestion
          => #loadProgram BYTES ~> #initVM ~> #precompiled?(ACCTCODE, SCHED) ~> #execute
```

**Comment by ehildenb** [comment]
@anvacaru please check through `driver.md` and `state-loader.md` and check for any other places that touch the `<program>` cell, to see if they need to be made to go through `#loadProgram` instead.

**Comment by anvacaru** [comment]
> @anvacaru please check through `driver.md` and `state-loader.md` and check for any other places that touch the `<program>` cell, to see if they need to be made to go through `#loadProgram` instead.

There are two `loadCallState` functions which alter the `<program>` cell in those files.
https://github.com/kframework/evm-semantics/blob/master/driver.md#261 @ line 261
https://github.com/kframework/evm-semantics/blob/master/state-loader.md @ line 148

**Comment by ehildenb** [comment]
@anvacaru it looks like both of those locations should also use the new `#loadProgram` primitive.

File: driver.md [hunk 3]
```diff
@@ -257,7 +257,7 @@ Note that `TEST` is sorted here so that key `"network"` comes before key `"pre"`
     rule <k> loadCallState { "origin" : (ORIG:Int), REST => REST }     ... </k> <origin> _ => ORIG     </origin>
     rule <k> loadCallState { "address" : (ACCTTO:Int), REST => REST }  ... </k> <id>     _ => ACCTTO   </id>
 
-    rule <k> loadCallState { "code" : (CODE:OpCodes), REST => REST} ... </k>
+    rule <k> loadCallState { "code" : (CODE:OpCodes), REST } => #loadProgram #asmOpCodes(CODE) ~> loadCallState { REST } ... </k>
          <program> _ => #asmOpCodes(CODE) </program>
          <jumpDests> _ => #computeValidJumpDests(#asmOpCodes(CODE)) </jumpDests>
```
Comment (ehildenb) [inline driver.md:262]:
```suggestion
    rule <k> loadCallState { "code" : (CODE:OpCodes), REST } => #loadProgram #asmOpCodes(CODE) ~> loadCallState { REST } ... </k>
```

File: state-loader.md [hunk 4]
```diff
@@ -144,7 +144,7 @@ Here we load the environmental information.
  // -------------------------------------
     rule <k> loadCallState { "data" : ( DATA:String => #parseByteStack( DATA ) ), _REST } ... </k>
 
-    rule <k> loadCallState { "code" : CODE:ByteArray, REST => REST } ... </k>
+    rule <k> loadCallState { "code" : CODE:ByteArray, REST } => #loadProgram CODE ~> loadCallState { REST } ... </k>
          <program> _ => CODE </program>
          <jumpDests> _ => #computeValidJumpDests(CODE) </jumpDests>
```
Comment (ehildenb) [inline state-loader.md:149]:
```suggestion
    rule <k> loadCallState { "code" : CODE:ByteArray, REST } => #loadProgram CODE ~> loadCallState { REST } ... </k>
```

**Review by ehildenb — APPROVED** [review]

**Review by ehildenb — APPROVED** [review]

**Pull request merged**